code/game/objects/structures/crates_lockers/crates.dm

```diff
@@ -20,6 +23,44 @@
 	drag_slowdown = 0
 	var/obj/item/paper/fluff/jobs/cargo/manifest/manifest
 
+	///counter for crates that push this one. Start from 1 and increment when Bumped.
+	var/bump_chain = 1
+
+/// When crate fail move to another place it call Bump(atom/A)
+/obj/structure/closet/crate/Bump(atom/A)
+	..()
+	///check that the crate can push the obstacle that is blocking it or not.

```

Comment:
you dont need to add comments like this, this isnt what the code docs requirement is